linker/Linker.Steps/OutputStep.cs

```diff
@@ -116,31 +116,38 @@ void OutputAssembly (AssemblyDefinition assembly)
 			var action = Annotations.GetAction (assembly);
 			Context.LogMessage (MessageImportance.Low, $"Output action: {action,8} assembly: {assembly}");
 
-			switch (action) {
-			case AssemblyAction.Save:
-			case AssemblyAction.Link:
-			case AssemblyAction.AddBypassNGen:
-				Context.Tracer.AddDependency (assembly);
-				WriteAssembly (assembly, directory);
-				break;
-			case AssemblyAction.Copy:
-				Context.Tracer.AddDependency (assembly);
-				CloseSymbols (assembly);
-				CopyAssembly (GetOriginalAssemblyFileInfo (assembly), directory, Context.LinkSymbols);
-				break;
-			case AssemblyAction.Delete:
-				CloseSymbols (assembly);
-				var target = GetAssemblyFileName (assembly, directory);
-				if (File.Exists (target)) {
-					File.Delete (target);
-					File.Delete (target + ".mdb");
-					File.Delete (Path.ChangeExtension (target, "pdb"));
-					File.Delete (GetConfigFile (target));
+			try
+			{
+				switch (action) {
+				case AssemblyAction.Save:
+				case AssemblyAction.Link:
+				case AssemblyAction.AddBypassNGen:
+					Context.Tracer.AddDependency (assembly);
+					WriteAssembly (assembly, directory);
+					break;
+				case AssemblyAction.Copy:
+					Context.Tracer.AddDependency (assembly);
+					CloseSymbols (assembly);
+					CopyAssembly (assembly, directory);
+					break;
+				case AssemblyAction.Delete:
+					CloseSymbols (assembly);
+					DeleteAssembly (assembly, directory);
+					break;
 				}
-				break;
-			default:
+			} finally {
```

Comment:
I am not sure I see any benefit of this. what was the intention to use it for ?

Reply:
Here's the scenario
* You start running all unit tests
* Tests are running fine
* Then one fails because of an exception thrown and Dispose doesn't get called on the SymbolReader
* All subsequent tests now fail because the temporary directory can't be cleaned up.

I didn't run into this with this code, but I did hit this scenario with changes coming to ResultChecker to verify the symbols.

Reply:
Sure but the symbols are already closed, e.g. at https://github.com/mono/linker/pull/245/files#diff-78e7cd35fdf21ba1a7924c0def208816R134

Reply:
Ok.  I'll undo this change.

Reply:
Done